assets/src/components/detours/detourMap.tsx

```diff
@@ -211,18 +211,44 @@ const OriginalRouteShape = ({
   children,
   classNames,
   onClick,
-}: OriginalRouteShapeProps) => (
-  <Polyline
-    positions={positions}
-    className={joinClasses([
-      "c-detour_map--original-route-shape",
-      ...classNames,
-    ])}
-    bubblingMouseEvents={false}
-    eventHandlers={{
-      click: onClick,
-    }}
-  >
-    {children}
-  </Polyline>
-)
+}: OriginalRouteShapeProps) => {
+  const [hovered, setHovered] = useState(false)
+
+  const eventHandlers = {
+    click: onClick,
+    mouseover: () => {
+      setHovered(true)
+    },
+    mouseout: () => {
+      setHovered(false)
+    },
+  }
+
+  return (
+    <>
+      <Polyline
+        key={`outline-hovered-${hovered}`}
```

Comment:
This `key` is here because Leaflet doesn't React that well - without it, Leaflet doesn't re-render the Polyline with the updated `opacity`. 

I'm not convinced that this is the best way to do this.